provider/pkg/clients/unstructured.go

```diff
@@ -138,6 +143,37 @@ func normalizeSecret(uns *unstructured.Unstructured) *unstructured.Unstructured
 	return uns
 }
 
+// normalizeSecretData normalizes the .data field of a Secret resource by trimming whitespace from string values.
+// This is necessary because the apiserver will trim whitespace from the .data field values, but the provider does not.
+func normalizeSecretData(uns *unstructured.Unstructured) *unstructured.Unstructured {
+	data, found, err := unstructured.NestedMap(uns.Object, "data")
+	if err != nil || !found || len(data) == 0 {
+		return uns
+	}
+
+	for k, v := range data {
+		if s, ok := v.(string); ok {
+			// Trim whitespace from the string value, for consistency with the apiserver which
+			// does the decoding and re-encoding to validate the value provided is valid base64.
+			// See: https://github.com/kubernetes/kubernetes/blob/41890534532931742770a7dc98f78bcdc59b1a6f/staging/src/k8s.io/apimachinery/pkg/runtime/codec.go#L212-L260
+			base64Decoded, err := base64.StdEncoding.DecodeString(s)
+			if err != nil {
+				// TODO: propagate error upwards to parent Normalize function to fail early. It is safe to
+				// ignore this error for now, since the apiserver will reject the resource if the value cannot
+				// be decoded.
+				continue
+			}
+			base64ReEncoded := base64.StdEncoding.EncodeToString(base64Decoded)
```

Comment:
Now decoding and re-encoding to base64 to be closer to what Kubernetes does instead of simply trimming trailing whitespace.

Reply:
Thanks for the explanation, it makes sense now.